### DISCUSSION TITLE
not kwh entities

### DISCUSSION THREAD

**Discussion by gpsmith** [WillCodeForCats/solaredge-modbus-multi#707] [Q&A] [answered]
I've started playing with HA and trying to get predbat installed. It wants three kwh sensors, but while I've managed to connect to the modbus, I am not getting any kwh entities from this integration.

I have a SE6000H HD wave setapp inverter, home backup interface and SE battery. the import and export modbus will be in backup interface.

These are the things predbat is looking for.  

- sensor.solaredge_i1_ac_energy_kwh
- sensor.solaredge_m1_imported_kwh
- sensor.solaredge_m1_exported_kwh

I'm expecting you to tell me the problem is between the chair and the keyboard. Thanks for your help.

**Comment by WillCodeForCats** [comment, marked as answer]
Sensor names won't have "kwh" in the names since that's a unit, and the units can be dynamically changed in the HA UI. Click on the sensor name and then the gear icon to see the entity name.

>sensor.solaredge_i1_ac_energy_kwh

This will be the "AC Energy" sensor in the inverter: `sensor.solaredge_i1_ac_energy`

<img width="575" alt="Screenshot 2024-12-02 at 14 19 55" src="https://github.com/user-attachments/assets/de6d63fa-054e-4b03-93da-9c35d4cc8ed1">

>sensor.solaredge_m1_imported_kwh
>sensor.solaredge_m1_exported_kwh

`sensor.solaredge_i1_m1_exported` and `sensor.solaredge_i1_m1_imported`

Energy exported or imported in the meter. Example:

<img width="581" alt="Screenshot 2024-12-02 at 14 19 16" src="https://github.com/user-attachments/assets/1bdc8e16-4433-4829-93a6-a834ef20daff">

**Reply by gpsmith** [reply]
This makes sense. I assumed I was missing them because there was no kwh named ones. I found these, which seem to provide the data, although they are not daily, but running total numbers. I've yet to work out if this caused predbat a problem. Once I've sorted a few other things, I guess I will find out. 

sensor.solaredge_i1_ac_energy
sensor.solaredge_i1_m1_ac_energy_imported
sensor.solaredge_i1_m1_ac_energy_exported

**Reply by WillCodeForCats** [reply]
Yes they are total incrementing. The inverter modbus interface does not provide daily values, but you can calculate daily value with a utility meter helper set to "daily" meter reset cycle.

**Reply by gpsmith** [reply]
Thanks for all your help. Really appreciated.

**Reply by Ashpork** [reply]
If it helps at all, my configuration for predbat is attached here. I also make use of predheat, but you should be able to ignore/delete that section.

[apps (1).yaml.txt](https://github.com/user-attachments/files/17990708/apps.1.yaml.txt)



My "daily" entities are essentially utility meters, making use of other entities which are calculating solar production vs. batt vs. grid etc.

```
utility_meter:
  solar_panel_production_daily:
    unique_id: solar_panel_production_daily
    source: sensor.solar_panel_production_kwh
    name: Solar Panel Production Daily
    cycle: daily
```
